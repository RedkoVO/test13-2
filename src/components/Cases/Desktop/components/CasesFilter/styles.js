import Searcher from '../../../../../assets/images/svg/searcher.svg'

export default () => ({
  root: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '720px',
    padding: '10px 25px',
    boxSizing: 'border-box',
    borderRadius: '25px',
    backgroundColor: 'rgba(0,0,0,0.08)'
  },
  casesFilterSearch: {
    width: '255px',
    height: '35px',
    padding: '15px 22px',
    color: '#fff',
    boxSizing: 'border-box',
    fontSize: '16px',
    fontWeight: 500,
    border: 'none',
    outline: 'none',
    backgroundColor: 'rgba(0,0,0,0.13)',
    borderRadius: '18px'
  },
  casesFilterOptions: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'space-between',
    height: '50px',
    width: '335px'
  },
  casesFilterType: {
    width: '100%',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  casesFilterItem: {
    fontSize: '11px',
    fontWeight: 500,
    padding: '5px',
    boxSizing: 'border-box',
    backgroundColor: 'rgba(0,0,0,0.13)',
    color: '#fff',
    textDecoration: 'none',

    '&:first-child': {
      borderTopLeftRadius: '15px',
      borderBottomLeftRadius: '15px',
      paddingLeft: '15px'
    },
    '&:last-child': {
      borderTopRightRadius: '15px',
      borderBottomRightRadius: '15px',
      paddingRight: '15px'
    }
  },
  casesFilterGameItem: {
    fontSize: '11px',
    fontWeight: 500,
    padding: '5px',
    boxSizing: 'border-box',
    backgroundColor: 'rgba(0,0,0,0.13)',
    color: '#fff',
    textDecoration: 'none'
  },
  selectedItem: {
    fontWeight: 700,
    backgroundColor: 'rgba(255,255,255,0.14)'
  },
  casesFilterBtn: {
    position: 'relative',
    width: '42px',
    height: '42px',
    background: `url(${Searcher})`,
    backgroundColor: ' rgba(0,0,0,0.13)',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    backgroundSize: '20px',
    borderRadius: '15px'
  }
})
