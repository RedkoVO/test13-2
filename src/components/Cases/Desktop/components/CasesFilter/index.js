import React from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'
import withStyles from '@material-ui/core/styles/withStyles'

import styles from './styles'

const CasesFilter = ({ classes }) => (
  <div className={classes.root}>
    <input type="text" className={classes.casesFilterSearch} placeholder="Search" />
    <div className={classes.casesFilterOptions}>
      <div className={classes.casesFilterType}>
        <div className={classes.casesFilterItem}>less than $15&lt;</div>
        <div className={classes.casesFilterItem}>$15-30</div>
        <div className={cn(classes.casesFilterItem, classes.selectedItem)}> $30-60</div>
        <div className={classes.casesFilterItem}>$60-90</div>
        <div className={classes.casesFilterItem}>less than $15&lt;</div>
      </div>
      <div className={classes.casesFilterType}>
        <div className={classes.casesFilterItem}>CS:GO</div>
        <div className={classes.casesFilterItem}>Dota 2</div>
        <div className={cn(classes.casesFilterItem, classes.selectedItem)}>Starcraft 2</div>
        <div className={classes.casesFilterItem}> Maincraft</div>
        <div className={classes.casesFilterItem}>Rainbow Six</div>
      </div>
    </div>
    <div className={classes.casesFilterBtn} />
  </div>
)

CasesFilter.propTypes = {
  classes: PropTypes.object
}

export default withStyles(styles)(CasesFilter)
