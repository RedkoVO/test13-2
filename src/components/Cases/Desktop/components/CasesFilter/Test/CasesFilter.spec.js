import * as React from 'react'
import renderer from 'react-test-renderer'

import CasesFilter from '../'

it('renders correctly', () => {
  const tree = renderer.create(<CasesFilter />).toJSON()
  expect(tree).toMatchSnapshot()
})
