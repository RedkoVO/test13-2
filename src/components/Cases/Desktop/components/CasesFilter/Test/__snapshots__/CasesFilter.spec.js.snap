// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="CasesFilter-root-1"
>
  <input
    className="CasesFilter-casesFilterSearch-2"
    placeholder="Search"
    type="text"
  />
  <div
    className="CasesFilter-casesFilterOptions-3"
  >
    <div
      className="CasesFilter-casesFilterType-4"
    >
      <div
        className="CasesFilter-casesFilterItem-5"
      >
        less than $15&lt;
      </div>
      <div
        className="CasesFilter-casesFilterItem-5"
      >
        $15-30
      </div>
      <div
        className="CasesFilter-casesFilterItem-5 CasesFilter-selectedItem-7"
      >
         $30-60
      </div>
      <div
        className="CasesFilter-casesFilterItem-5"
      >
        $60-90
      </div>
      <div
        className="CasesFilter-casesFilterItem-5"
      >
        less than $15&lt;
      </div>
    </div>
    <div
      className="CasesFilter-casesFilterType-4"
    >
      <div
        className="CasesFilter-casesFilterItem-5"
      >
        CS:GO
      </div>
      <div
        className="CasesFilter-casesFilterItem-5"
      >
        Dota 2
      </div>
      <div
        className="CasesFilter-casesFilterItem-5 CasesFilter-selectedItem-7"
      >
        Starcraft 2
      </div>
      <div
        className="CasesFilter-casesFilterItem-5"
      >
         Maincraft
      </div>
      <div
        className="CasesFilter-casesFilterItem-5"
      >
        Rainbow Six
      </div>
    </div>
  </div>
  <div
    className="CasesFilter-casesFilterBtn-8"
  />
</div>
`;
